src/Table/Table.tsx

```diff
@@ -0,0 +1,809 @@
+                        }
+                        return originalSubrow;
+                    });
+                    return { ...originalRow, subRows: updatedSubrows, selected: false };
+                }
+            }
+            return originalRow;
+        });
+
+        return updatedRows;
+    };
+
+    const setDefaultTableRows = async () => {
+        const updatedRows: Array<TableRow> = await getRows(props.data);
+        setTableRows(updatedRows);
+        setTableRowsImage(updatedRows);
+    };
+
+    const doPaginate = (): void => {
+        if (props.usePagination && props.currentpage && props.offset && (tableRows.length > 0)) {
+            // pagination start from 1 hence the need fro deducting 1
+            const start: number = (props.currentpage - 1) * props.offset;
+            const end: number = (props.offset * (props.currentpage));
+
+            const currentPage: Array<TableRow> = tableRows.slice(start, end);
+            setCurrentTableRows(currentPage);
+        } else {
+            setCurrentTableRows(tableRows);
+        }
+    };
+
+    const RowsAreCollapsable = (): boolean => {
+        return currentTableRows.some((row: TableRow) => {
+            return (
+                ((row.subRows.length > 0) || row.rowContentDetail) ||
+                (row.subRows.some((subRow: TableRow) => subRow.rowContentDetail || (row.subRows && row.subRows.length > 0)))
+            );
+        }) && !!props.onRowExpanded;
+    };
+
+    const doSearch = (): void => {
+        let searchResult: Array<TableRow> = [];
+        if (props.searchProps?.searchText && props.searchProps?.searchInColumns && props.searchProps?.searchInColumns.length > 0) {
+            searchResult = searchTextInArray(tableRowsImage, props.searchProps.searchText, props.searchProps.searchInColumns);
+        } else {
+            searchResult = [...tableRowsImage];
+        }
+
+        setTableRows(searchResult);
+        props.searchProps?.onSearch && props.searchProps?.onSearch(searchResult);
+    };
+
+    // useEffects ----------------------------------------------------------
+
+    React.useEffect(() => {
+        doPaginate();
+    }, [tableRows]);
+
+    React.useEffect(() => {
+        if (!!props.onRowSelected) {
+            const notAllsAreRowsSelected = tableRows.some((row: TableRow) => !row.selected);
+
+            if (notAllsAreRowsSelected) {
+                setAllRowsChecked(false);
+            } else {
+                setAllRowsChecked(true);
+            }
+        }
+    }, [currentTableRows]);
+
+    React.useEffect(() => {
+        if (!!props.searchProps?.onSearch) {
+            if (props.searchProps.triggerSearchOn === "Change") {
+                doSearch();
+            }
+        }
+    }, [props.searchProps?.searchInColumns, props.searchProps?.searchText]);
+
+    React.useEffect(() => {
+        if (!!props.searchProps?.onSearch) {
+            if (props.searchProps?.triggerSearchOn === "Submit") {
```

Comment:
i think u can just use one if with && here

Reply:
They are individual checking kp. ` `props.searchProps?.triggerSearchOn === "Submit"` is the same as props.searchProps && props.searchProps.triggerSearchOn === "Submit" `

Reply:
i think what she meant is 
```suggestion
            if (!!props.searchProps?.onSearch && props.searchProps?.triggerSearchOn === "Submit") { doSearch() }
```